docs/sphinx/source/contributing.rst

```diff
@@ -0,0 +1,24 @@
+.. _contributing:
+
+Contributing
+==================
+
+Code
+------------------
+
+Please submit code patches through the bpf mailing list, bpf@vger.kernel.org. You can find the archive `here <https://lore.kernel.org/bpf/>`_.
+More on using the linux mailing list system can be found `here <https://www.kernel.org/doc/html/v4.17/process/submitting-patches.html>`_.
+
+
+Documentation
+------------------
+
+This documentation is generated using `sphinx <https://www.sphinx-doc.org/en/master/>`_. The API documentation specifically is generated using `doxygen <https://www.doxygen.nl/index.html>`_ and the sphinx plugin, `breathe <https://breathe.readthedocs.io/en/latest/>`_.
```

Comment:
general suggestion -- can you wrap all the lines at 78 characters? It will be easier to read and review source text, and it shouldn't matter for the final output.

Reply:
> So first of all, thanks a lot for sticking with this and figuring out how to do it! It looks much simpler than I thought it would turn out, so kudos for that as well! :)
> 
> Now overall questions and concerns.
> 
> 1. I wonder if we should put all the .rst files into kernel repo, next to the original source of libbpf. That way more people will be able to review it and not everything will have to be blocked on me. The downside is that it will be harder for casual contributors to contribute, because they would need to get through all the horrors of setting up kernel development environment, figuring out git send-email settings, all the conventions, etc, etc. I wonder what your thoughts are?

Would there be any issue with getting these files merged upstream to other kernel trees? If not i'd actually be in favor of keeping them in kernel tree.

I do agree with your assessment of tradeoffs. I see libbpf as a great entry point for people who want to get into kernel development but don't know where to start and would want to create some blog posts/videos showing how to submit patches and demonstrate documentation as a way of getting into the kernel world. The Go team did something very similar for documenting the standard library and it was hugely successful.  

Plus the code comment documentation would have to be submitted through the mailing list anyway, so perhaps it's best to keep it all in one spot anyway.

> 2. Do docs need to leave deep under docs/sphinx/source? I wonder if it will be nicer if all the documentation will be directly under docs/, while all the support configs and whatever else necessary to do generation would go under scripts/ (or we can create tools/)?

I think that can all be rewired, I will try to do that this week. 

> 3. How does the integration with Github Actions and readthedocs look like in general terms? Who's pulling/pushing updates, etc?

There's actually no setup on our end. Readthedocs monitors the github repo for changes and builds accordingly. No github actions required. I used github actions for the previous approach of hosting on netlify.